docs/4.2/production.md

```diff
@@ -0,0 +1,269 @@
+# Production Guide
+
+This guide provides more in-depth details for running Teleport in Production. 
+
+<!-- TODO: Minimal Config example -->
+
+[TOC]
+
+## Prerequisites
+
+* Read the [Architecture Overview](architecture/teleport_architecture_overview.md)
+* Read through the [Installation Guide](installation.md) to see the available packages and binaries available.
+* Read the CLI Docs for [`teleport`](cli-docs.md#teleport)
+
+## Designing Your Cluster
+
+Before installing anything there are a few things you should think about.
+
+* Where will you host Teleport?
+    * On-premises
+    * Cloud VMs such as [AWS EC2](aws_oss_guide.md) or [GCE](gcp_guide.md)
+    * An existing Kubernetes Cluster
+* What does your existing network configuration look like?
+    * Are you able to administer the network firewall rules yourself or do you need to work with a network admin?
+    * Are these nodes accessible to the public Internet or behind NAT?
+* Which users ([Roles or ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) on k8s) are set up on the existing system?
+   * Can you add new users or Roles yourself or do you need to work with a system administrator ?
+
+## Firewall Configuration
+
+Teleport services listen on several ports. This table shows the default port numbers.
+
+|Port      | Service    | Description | Ingress | Egress
+|----------|------------|-------------|---------|----------
+| 3080      | Proxy      | HTTPS port clients connect to. Used to authenticate `tsh` users and web users into the cluster. | Allow inbound connections from HTTP and SSH clients.| Allow outbound connections to HTTP and SSH clients.
+| 3023      | Proxy      | SSH port clients connect to after authentication. A proxy will forward this connection to port `3022` on the destination node. | Allow inbound traffic from SSH clients. | Allow outbound traffic to SSH clients.
+| 3022      | Node       | SSH port to the Node Service. This is Teleport's equivalent of port `22` for SSH. | Allow inbound traffic from proxy host. | Allow outbound traffic to the proxy host.
+| 3025      | Auth       | SSH port used by the Auth Service to serve its Auth API to other nodes in a cluster. | Allow inbound connections from all cluster nodes. | Allow outbound traffic to cluster nodes.
+| 3024      | Proxy      | SSH port used to create "reverse SSH tunnels" from behind-firewall environments into a trusted proxy server. | <TODO> | <TODO>
+
+<!--TODO: Add several diagrams of firewall config examples-->
+
+## Installation
+
+We have a detailed [installation guide](/installation.md) which shows how to
+install all available binaries or [install from
+source](#../installation.md#installing-from-source). Reference that guide to learn
```

Comment:
Two different styles of link pointing to the same file. I think we should probably use the relative style (`../installation.md`) for both.